tapback/backend.c

```diff
@@ -262,6 +292,95 @@ device_set_mode(vbd_t *device, const char *mode) {
 	return 0;
 }
 
+int
+physical_device_path_changed(vbd_t *device) {
+    int err = 0, minor = -1;
+    pid_t pid = -1;
+    char *s = NULL;
+
+    /*
+     * Assume there will never be an OS with pid_t larger than long.
+     */
+    const char *nbd_pattern = "/run/blktap-control/nbd%ld.%d";
+    device->tap = NULL;
+
+    s = tapback_device_read(device, XBT_NULL, PHYS_DEV_PATH_KEY);
+    if (!s) {
+        err = -errno;
+        WARN(device, "Failed to read the %s: %s\n",
+             PHYS_DEV_PATH_KEY, strerror(-err));
+        goto done;
+    }
+
+    if (sscanf(s, nbd_pattern, &pid, &minor) != 2) {    
+        err = -EINVAL;
+     	WARN(device, "Malformed %s: '%s'\n", PHYS_DEV_PATH_KEY, s);
+        goto out;
+    }
+
+    device->tap = malloc(sizeof(*device->tap));
+    if (!device->tap) {
+        err = -ENOMEM;
+        goto out;
+    }
+
+    err = find_tapdisk_by_pid(pid, device->tap);
+    if (err) {
+        WARN(device, "Error looking for tapdisk: %s\n", strerror(-err));
+        goto out;
+    }
+    INFO(device, "Found tapdisk[%d]\n", device->tap->pid);
+    device->minor = minor;
+
+    /*
+     * get the VBD parameters from the tapdisk
+     */
+    if ((err = tap_ctl_info(device->tap->pid, &device->sectors,
+                    &device->sector_size, &device->info,
+                    device->minor))) {
+        WARN(device, "error retrieving disk characteristics: %s\n",
+             strerror(-err));
+        goto out;
+    }
```

Comment:
This looks like it has a mix of tabs and spaces. blktap should follow the kernel coding standards and be all tabs (8 char). There are places where this isn't the case but new code should comply.

Reply:
fixed